components/frontend/src/utils.jsx

```diff
@@ -104,12 +105,34 @@ export function isMeasurementOutdated(metric) {
     if (metric.latest_measurement) {
         return metric.latest_measurement.outdated ?? false
     }
-    return Object.keys(metric.sources ?? {}).length > 0 // If there are sources, measurement is needed
+    return false
 }
 isMeasurementOutdated.propTypes = {
     metric: metricPropType,
 }
 
+export function isSourceConfigurationComplete(dataModel, metric) {
+    // Return whether the metric can be measured, meaning that there is at least one source and all sources have
+    // all mandatory parameters configured
+    const sources = Object.values(metric.sources ?? {})
+    if (sources.length === 0) {
+        return false
+    }
+    return sources.every((source) => {
+        const parameters = dataModel.sources[source.type].parameters
+        return Object.entries(parameters).every(([parameterKey, parameter]) => {
+            if (parameter.mandatory && parameter.metrics.includes(metric.type) && !parameter.default_value) {
+                return source?.parameters?.[parameterKey]
```

Comment:
Why return the value here? Do we ever return `false`?

Reply:
This is how `every` works, it takes a callback for each item that should return true or false, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every